### PR TITLE
Add Jackson deserialization support

### LINKED ISSUE #211: First-class support for Jackson
Something like:
`responseJson<T>(handler: (Request, Response, Result<T, FuelError>) -> Unit)`, where `T` can be `Json` or a Jackson type like `JsonNode`.

### PR DESCRIPTION
**Pull request by deiga**
Fixes #211

### CODE REVIEW

**Comment by deiga** [comment]
@yoavst It does, it's nearly an exact copy of the implementation :)

**Comment by yoavst** [comment]
Can you add a test with list deserialization?

**Comment by deiga** [comment]
@yoavst There already is one, it's the last in the file :)

**Comment by deiga** [comment]
Does anyone know how to fix code coverage? :(

**Comment by yoavst** [comment]
@deiga Add one more to test to keep the coverage above the old coverage and it is OK for me (also now, but let's keep @codecov happy)

**Comment by deiga** [comment]
@yoavst Any idea why the Android tests suddenly started failing?

**Comment by deiga** [comment]
@yoavst Any idea why the Android tests suddenly started failing?

**Comment by yoavst** [comment]
No idea :(

File: fuel-jackson/build.gradle [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+repositories {
+    jcenter()
+}
+
+dependencies {
+    compile "com.fasterxml.jackson.module:jackson-module-kotlin:$rootProject.jackson_version"
+    compile "org.jetbrains.kotlin:kotlin-stdlib:$parent.ext.kotlin_version"
+    compile project(':fuel')
+    testCompile "junit:junit:$parent.ext.junit_version"
+    testCompile group: 'org.hamcrest', name: 'hamcrest-all', version: '1.3'
```
Comment (kittinunf) [inline fuel-jackson/build.gradle]: I think `hamcrest` is not needed...

File: fuel-jackson/build.gradle [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+jacocoTestReport {
+    reports {
+        html.enabled = false
+        xml.enabled = true
+        csv.enabled = false
+    }
+}
+
+publish {
+    artifactId = 'fuel-jacskon'
```
Comment (kittinunf) [inline fuel-jackson/build.gradle]: Typos here

File: fuel-jackson/build.gradle (hunk 1 above)
Comment (deiga) [reply fuel-jackson/build.gradle]: @kittinunf Well, hamcrest is already included with junit. This just adds a bunch of Matchers. But, sure I'll remove it.

**Comment by kittinunf** [comment]
```
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':fuel-android:testDebugUnitTest'.
```

Ah, this is weird. I don't think the test is failing on master branch. At first glance, I do not see anything wrong in here. 🤔

**Comment by deiga** [comment]
Even more weird is that it doesn't fail locally.

**Comment by kittinunf** [comment]
Can you help me rebase against the last master? Thanks!

**Comment by deiga** [comment]
@kittinunf Done! :)

**Comment by kittinunf** [comment]
This looks great to me! Thanks @deiga. 👍 

@yoavst do you have anything to shout out before we merge this?

**Comment by yoavst** [comment]
Not :+1:

**Pull request merged**